src/Tasks/Copy.cs

```diff
@@ -29,14 +51,23 @@ public Copy()
         #region Properties
 
         private bool _canceling;
-        private readonly HashSet<string> _directoriesKnownToExist = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+        // Bool is just a placeholder, we're mainly interested in a threadsafe key set.
+        private readonly ConcurrentDictionary<string, bool> _directoriesKnownToExist = new ConcurrentDictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
 
         /// <summary>
         /// Force the copy to retry even when it hits ERROR_ACCESS_DENIED -- normally we wouldn't retry in this case since 
         /// normally there's no point, but occasionally things get into a bad state temporarily, and retrying does actually 
         /// succeed.  So keeping around a secret environment variable to allow forcing that behavior if necessary.  
         /// </summary>
-        private static bool s_alwaysRetryCopy = Environment.GetEnvironmentVariable("MSBUILDALWAYSRETRY") != null;
+        private static bool s_alwaysRetryCopy = Environment.GetEnvironmentVariable(AlwaysRetryEnvVar) != null;
+
+        /// <summary>
+        /// Global flag to force on UseSymboliclinksIfPossible since Microsoft.Common.targets doesn't expose the functionality.
+        /// </summary>
+        private static readonly bool s_forceSymlinks = Environment.GetEnvironmentVariable("MSBuildUseSymboliclinksIfPossible") != null;
```

Comment:
?

https://github.com/Microsoft/msbuild/blob/8c2c86bd5ca84b8fd6f6e555111dd55d750024d7/src/Tasks/Microsoft.Common.CurrentVersion.targets#L4180

and several others. Is it missing somewhere? Should we just add it there?

Reply:
I was unsure what to do here: Was it a deliberate choice to avoid exposing this functionality? It's not in the .targets so I figured to make it reachable by env var. I can pull this from this PR as well, as I have an experiment to do after this PR to try ReFS copy-on-write Clones instead of File.Copy(), as a Windows-specific flag and functionality, and would be touching this area again.